packages/design-system/src/components/Modal/Modal.tsx

```diff
@@ -64,53 +69,61 @@ function Modal(props: ModalPropsType): ReactElement {
 	}
 
 	return (
-		<DialogBackdrop {...dialog} className={styles['modal-backdrop']} data-test="modal.backdrop">
-			<Dialog
-				{...dialog}
-				data-test="modal"
-				className={styles['modal']}
-				hide={preventEscaping ? undefined : () => onClose()}
-				ref={ref}
-			>
-				<StackVertical gap={0}>
-					<div className={styles['modal__header']}>
-						{header.icon && <ModalIcon icon={header.icon} data-test="modal.header.icon" />}
-						<div className={styles['modal-header-text']}>
-							<span className={styles['modal-header-text__title']} data-test="modal.header.title">
-								{header.title}
-							</span>
-							{header.description && (
-								<span
-									className={styles['modal-header-text__description']}
-									data-test="modal.header.description"
-								>
-									{header.description}
-								</span>
-							)}
-						</div>
-					</div>
+		<>
+			{disclosure && disclosure(() => dialog.setVisible(true))}
```

Comment:
Interesting change ☝️

Reply:
Mostly indent 👇

Reply:
There's more to the disclosure than the function though: reakit will spread attributes on the disclosure to let users know that it's connected to a modal.